src/main/java/org/spongepowered/common/data/persistence/JsonDataFormat.java

```diff
@@ -211,7 +212,29 @@ private static void write(JsonWriter writer, @Nullable Object value) throws IOEx
         } else if (value instanceof String) {
             writer.value((String) value);
         } else if (value instanceof Iterable) {
-            writeArray(writer, (Iterable<?>) value);
+            writer.beginArray();
+            for (Object item : (Iterable<?>) value) {
+                write(writer, item);
+            }
+            writer.endArray();
+        } else if (value instanceof byte[]) {
+            writer.beginArray();
+            for (Object item : (byte[]) value) {
+                write(writer, item);
+            }
```

Comment:
```java
for (byte item : (byte[]) value) {
    writer.write(item);
}
```
This would probably improve performance as it avoid boxing, recursion and instanceof.
Same goes for `int[]` and `long[]` below.

Reply:
Thanks, not sure why I was still using `Object` in there.